src/interface.jl

```diff
@@ -0,0 +1,252 @@
+
+Broadcast.broadcastable(b::Bijector) = Ref(b)
+
+"Computes the log(abs(det(J(x)))) where J is the jacobian of the transform."
+logabsdetjac(b::T1, y::T2) where {T<:Bijector,T1<:Inversed{T},T2} = 
+    error("`logabsdetjac(b::$T1, y::$T2)` is not implemented.")
+
+"Transforms the input using the bijector."
+transform(b::T1, y::T2) where {T<:Bijector,T1<:Inversed{T},T2} =
+    error("`transform(b::$T1, y::$T2)` is not implemented.")
+
+"Computes both `transform` and `logabsdetjac` in one forward pass."
+forward(b::T1, y::T2) where {T<:Bijector,T1<:Inversed{T},T2} = 
+    error("`forward(b::$T1, y::$T2)` is not implemented.")
+
+
+transform(b::Bijector) = x -> transform(b, x)
+(ib::Inversed{<: Bijector})(y) = transform(ib, y)
+
+# default `forward` implementations; should in general implement efficient way
+# of computing both `transform` and `logabsdetjac` together.
+forward(b::Bijector, x) = (rv=transform(b, x), logabsdetjac=logabsdetjac(b, x))
+forward(ib::Inversed{<: Bijector}, y) = (rv=transform(ib, y), logabsdetjac=logabsdetjac(ib, y))
+
+# defaults implementation for inverses
+logabsdetjac(ib::Inversed{<: Bijector}, y) = - logabsdetjac(ib.orig, transform(ib, y))
+
+inv(b::Bijector) = Inversed(b)
+inv(ib::Inversed{<:Bijector}) = ib.orig
+
+# AD implementations
+function jacobian(b::ADBijector{<: ForwardDiffAD}, y::Real)
+    ForwardDiff.derivative(z -> transform(b, z), y)
+end
+function jacobian(b::Inversed{<: ADBijector{<: ForwardDiffAD}}, y::Real)
+    ForwardDiff.derivative(z -> transform(b, z), y)
+end
+function jacobian(b::Inversed{<: ADBijector{<: ForwardDiffAD}}, y::AbstractVector{<: Real})
+    ForwardDiff.jacobian(z -> transform(b, z), y)
+end
+
+function jacobian(b::ADBijector{<: TrackerAD}, y::Real)
+    Tracker.gradient(z -> transform(b, z), y)[1]
+end
+function jacobian(b::Inversed{<: ADBijector{<: TrackerAD}}, y::Real)
+    Tracker.gradient(z -> transform(b, z), y)[1]
+end
+function jacobian(b::Inversed{<: ADBijector{<: TrackerAD}}, y::AbstractVector{<: Real})
+    Tracker.jacobian(z -> transform(b, z), y)
+end
+
+# TODO: allow batch-computation, especially for univariate case?
+"Computes the absolute determinant of the Jacobian of the inverse-transformation."
+function logabsdetjac(b::ADBijector, x::Real)
+    log(abs(jacobian(b, x)))
+end
+function logabsdetjac(b::ADBijector, x::AbstractVector{<:Real})
+    fact = lu(jacobian(b, x), check=false)
+    issuccess(fact) ? log(abs(det(fact))) : -Inf # TODO: or smallest possible float?
+end
+
+###############
+# Composition #
+###############
+
+struct Composed{A} <: Bijector
+    ts::A
+end
+
+compose(ts...) = Composed(ts)
+
+# The transformation of `Composed` applies functions left-to-right
+# but in mathematics we usually go from right-to-left; this reversal ensures that
+# when we use the mathematical composition ∘ we get the expected behavior.
+# TODO: change behavior of `transform` of `Composed`?
+∘(b1::B1, b2::B2) where {B1 <: Bijector, B2 <: Bijector} = compose(b2, b1)
```

Comment:
1. Style: `∘(b1::Bijector, b2::Bijector)` please :)
2. I completely agree with the composition order. Also agree that maybe we want to change the order of stuff in `Composed`.

Reply:
1. Fixed, also fixed knowledge. Was left-over from me not properly understanding the type-system:)
2. :+1: